covariance/covarianceXsec.cpp

```diff
@@ -425,144 +701,43 @@ void covarianceXsec::scanParameters() {
-	bool haskinbounds=false;
-
-	////////////////////
-	//New generic cuts things 
-	////////////////////
-	
-	//Only consider the kinematic string and the boundaries if you've actually given it a string to use...
-	if( ((TObjString*)xsec_kinematic_type->At(norm_counter))->GetString().Length() > 0){ 
-	  haskinbounds = true;
-	  //ETA - This can be a vector :) can provide different kinematic variables to cut on
-	  tmp_xsec.KinematicVarStr.push_back(std::string(((TObjString*)xsec_kinematic_type->At(norm_counter))->GetString()));
-
-	  //ETA - This can be a vector :) can provide different kinematic variables to cut on
-	  std::vector<double> single_selec;
-
-	  //ETA - push back kinematic type with dummy -999 since this needs to be converted into an enum for a kinematic type within
-	  //a samplePDFFD daughter class
-	  single_selec.push_back((*xsec_kinematic_lb)(i));
-	  single_selec.push_back((*xsec_kinematic_ub)(i));
-	  tmp_xsec.Selection.push_back(single_selec);
-
-	}
-	else{
-	  std::cout << "~~~" << std::endl;
-	  std::cout << "Did not find kinematic bound for param " << i << std::endl;
-	  std::cout << "~~~" << std::endl;
-	}
+	if ((GetXsecParamDetID(i) & 24) == 24) {//Far pars
 
-	tmp_xsec.hasKinBounds=haskinbounds;
-	//End of kinematic bound checking	
+	  // Now check if it's a spline parameter or not
+	  //This needs to be updated to check against a string
+	  if (strcmp(GetXsecParamType(i), "Spline") == 0) {//FarSplinePars
+		std::cout << GetXsecParamType(i) << std::endl;
+		FarSplineParsNames.push_back(GetParameterName(i));
+		FarSplineParsIndex.push_back(i);
 
-	// Set the global parameter index of the normalisation parameter
-	tmp_xsec.index=i;
-	//Add this parameter to the vector of parameters
-	FarNormParams.push_back(tmp_xsec);
-	nFarNormParams++;
+		for(int Mode_i = 0 ; Mode_i < _fFDSplineModes[nFarSplineParams].size() ; ++Mode_i){
+		  std::cout << "Mode to apply to is " << _fFDSplineModes[nFarSplineParams][Mode_i] << std::endl;
+		}
 
-	if (!(GetXSecParamID(i, 1) & 1)) {//non-Near affecting Far affecting norm parameters
-	  nFaronlyNormParams++;
-	}
-      }//End FarNormPars
-      else if (GetXSecParamID(i, 0) == -2) {//Far functional parameter
-        nFarFuncParams++;
-        FarFuncParsNames.push_back(GetParameterName(i));
-        FarFuncParsIndex.push_back(i);
-      }//End Far funcpars
-    }//End Far pars
+		FarSplineModes.push_back(_fFDSplineModes[nFarSplineParams]);
+		nFarSplineParams++;
 
-    //Near affecting parameters
-    if ((GetXSecParamID(i, 1) & 1)) {//Near affecting parameters
-      if (GetXSecParamID(i, 0) == -1) {//Near affecting norm pars
+		// Or a normalisation parameter
+	  } //End FarSplinePars
+	  else if (strcmp(GetXsecParamType(i), "Norm") == 0) {//FarNormPars
 		XsecNorms4 tmp_xsec;
+		tmp_xsec.name=GetParameterName(i);
 
-        tmp_xsec.name=GetParameterName(i);
+		tmp_xsec.modes=_fNormModes[i];
```

Comment:
Should rather be `tmp_xsec.modes=_fNormModes[norm_counter];` I think

Reply:
yep I've fixed this :)